### DISCUSSION TITLE
Modifying the logout URL

### DISCUSSION THREAD

**Discussion by ksgrg** [snipe/snipe-it#14402] [Q&A]
Hi Community,
Good Afternoon

Currently once a user is logged in the logout URL is https://asset.mydomain.com/logout I want this to be set to something else. I couldn't find any settings that would change this.

Could anyone redirect me to where the change can be made, please?

Thank you in advance

KS

**Comment by ksgrg** [comment]
Any suggestion from anyone ?

**Comment by marcusmoore** [comment]
There isn't a setting in the UI for this but you should be able to modify [this line](https://github.com/snipe/snipe-it/blob/8bb8eab69b062db4561ccae1ab32dea410d2dea3/routes/web.php#L523) and [this line](https://github.com/snipe/snipe-it/blob/8bb8eab69b062db4561ccae1ab32dea410d2dea3/routes/web.php#L528) in the route configuration code. 

Keep in mind you'll need to revert that change before upgrading (so the patches can be applied cleanly) and re-apply after the process. 

Also, your title mention "for a student" and this change would be application wide and not per-user.

May I ask why you want to modify that url?

**Comment by snipe** [comment]
I think you want this setting under `Admin Settings > Security`?

<img width="991" alt="Screenshot 2024-03-11 at 7 14 47 PM" src="https://github.com/snipe/snipe-it/assets/197404/aea01a64-445f-4258-afc3-266125d66b80">

**Comment by ksgrg** [comment]
> There isn't a setting in the UI for this but you should be able to modify [this line](https://github.com/snipe/snipe-it/blob/8bb8eab69b062db4561ccae1ab32dea410d2dea3/routes/web.php#L523) and [this line](https://github.com/snipe/snipe-it/blob/8bb8eab69b062db4561ccae1ab32dea410d2dea3/routes/web.php#L528) in the route configuration code.
> 
> Keep in mind you'll need to revert that change before upgrading (so the patches can be applied cleanly) and re-apply after the process.
> 
> Also, your title mention "for a student" and this change would be application wide and not per-user.
> 
> May I ask why you want to modify that url?

Hi [marcusmoore](https://github.com/marcusmoore)
Thank you for your response. We have SSO configured with UPN in Azure. What I was trying to achieve is they do not sign out at all but when they clicked on the logout URL they would be taken to a dashboard link otherwise it requires them to sign in with their email again. 

At the moment when they visit  https://asset.mydomain.com it logs them automatically. So is there a way we could change the logout URL to not logout the session but take them to the dashboard URL


> I think you want this setting under `Admin Settings > Security`?
> 
> <img alt="Screenshot 2024-03-11 at 7 14 47 PM" width="991" src="https://private-user-images.githubusercontent.com/197404/311838298-aea01a64-445f-4258-afc3-266125d66b80.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTAyMzI4MzksIm5iZiI6MTcxMDIzMjUzOSwicGF0aCI6Ii8xOTc0MDQvMzExODM4Mjk4LWFlYTAxYTY0LTQ0NWYtNDI1OC1hZmMzLTI2NjEyNWQ2NmI4MC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMzEyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDMxMlQwODM1MzlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1kYzIyZWYxYzAyZGEzYTk2YjA4YWJiN2FhOWQ0YjMzZDlhOTQyYjY4YzEyNzUwYTQ5OWY3MzQ1OWM3NGJiNmJkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.8sf7jaesqzXJCRxoEBhSnKfr1fmlVC40TFAgUY-_I2Y">

Hi [snipe](https://github.com/snipe)
Thank you for your reply. It did the job but it logs them out and they are required to login with their Office 365 account when they visit https://asset.mydomain.com.

Or is there a way we can completely remove the logout link from the user menu. I would remove the both change user and logout URL just for the user side completely.

![image](https://github.com/snipe/snipe-it/assets/162708626/8f2ef0d1-7fa7-481c-bd14-a82d9281d53d)

**Reply by marcusmoore** [reply]
> Or is there a way we can completely remove the logout link from the user menu. I would remove the both change user and logout URL just for the user side completely.

@ksgrg we don't have a way to change the login url to link them to a different page but to accomplish the removal of the "Change Password" and "Logout" links from the menu you can delete the following lines:

https://github.com/snipe/snipe-it/blob/9475871edb1e48d0613c8c509f8f0d0dcc988edc/resources/views/layouts/default.blade.php#L361-L366

https://github.com/snipe/snipe-it/blob/9475871edb1e48d0613c8c509f8f0d0dcc988edc/resources/views/layouts/default.blade.php#L377-L390

Keep in mind the caveat about having to revert the changes before updating is still valid.

**Comment by ksgrg** [comment]
Hi [marcusmoore](https://github.com/marcusmoore) 
Thank you for your response and the wonderful help.

The only reason I wanted to remove them was we have a LDAP Sync and SAML with Azure configured. When the user log out we want them to redirect to a certain page which it does now after making the following change from [snipeit](https://github.com/snipe)

![image](https://github.com/snipe/snipe-it/assets/162708626/60fd4541-e32c-40a5-b748-db4b9eb63bea)

The only thing is it logs them out of the Azure login session as well and when they visit the https://asset.mydomain.com page they have to login again which is something we want not to happen.
